nym-api/src/node_describe_cache/mod.rs

```diff
@@ -112,28 +115,28 @@ async fn try_get_client(
     }
 
     Err(NodeDescribeCacheError::NoHttpPortsAvailable {
-        host: gateway_host.clone(),
-        gateway: gateway.identity_key.clone(),
+        host: host.to_string(),
+        gateway: identity_key.to_string(),
     })
 }
 
-async fn get_gateway_description(
-    gateway: Gateway,
+async fn try_get_description(
+    data: RefreshData,
 ) -> Result<(IdentityKey, NymNodeDescription), NodeDescribeCacheError> {
-    let client = try_get_client(&gateway).await?;
+    let client = try_get_client(&data.host(), &data.identity_key(), data.port()).await?;
 
     let host_info =
         client
             .get_host_information()
             .await
             .map_err(|err| NodeDescribeCacheError::ApiFailure {
-                gateway: gateway.identity_key.clone(),
+                gateway: data.identity_key().to_string(),
                 source: err,
             })?;
 
```

Comment:
fyi. while I don't think I've put it on the client (because I didn't need it at the time), you could query for `self.get_json_from(routes::api::v1::roles_absolute()).await` to get info whether the node is a mixnode/gateway/runs IPR/NR from the node itself

Reply:
If we could then go back to contract cache, that would be a win, would that be another http request towards the node then, if so I'd rather avoid it, I'm worried about how long this will take as is...